### PR TITLE
samba: fix cross compilation

### PR DESCRIPTION
**Pull request by Myaats**
## Description of changes

Fix the regression of  `pkgsCross.aarch64-multiplatform.samba` failing building after #269643.

TL;DR: Seems like overriding the python derivation in nativeBuildInputs broke the buildPackages substitution. However it is not needed for the fix in the previous PR.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux (pkgsCross)
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Myaats** [comment]
@Mindavi Able to confirm the fix?

**Comment by Mindavi** [comment]
Checking now, thanks for looking into this!

**Comment by Mindavi** [comment]
It seemed to start building but then I needed to go do other things. At least it gets a lot further from what I've seen, but I didn't check if it finishes and if the results are all for aarch64. I can only look tomorrow evening (~UTC 20:00) at that

File: pkgs/servers/samba/4.x.nix [hunk 1]
```diff
@@ -223,7 +223,7 @@ stdenv.mkDerivation rec {
   '';
 
   disallowedReferences =
-    lib.optionals (buildPackages.python3Packages.python != python)
+    lib.optionals (buildPackages.python3Packages.python != python3Packages.python)
       [ buildPackages.python3Packages.python ];
```
Comment (lopsided98) [inline pkgs/servers/samba/4.x.nix:227]: Rather than reverting this change, it might make sense to extend it to verify that we don't end up with two copies of Python (regular and libxcrypt-legacy) in the closure:
```suggestion
    lib.optional (buildPackages.python3Packages.python != python3Packages.python)
      buildPackages.python3Packages.python ++
    lib.optional (python3Packages.python != python) python3Packages.python;
```
```

**Review by lopsided98 — COMMENTED** [review]
Tested cross-compiling to armv6l successfully.

File: pkgs/servers/samba/4.x.nix (hunk 1 above)
Comment (lopsided98) [reply pkgs/servers/samba/4.x.nix:227]: Actually, if you try this you will discover that samba does have multiple Pythons in its closure. Fixing that might be out of scope for this PR though.

**Review by Mindavi — APPROVED** [review]
LGTM, and I don't see any native (build platform) binaries in the closure, so I think that's good enough for now.

**Comment by Mindavi** [comment]
I think further improvements can be done in follow-ups, if necessary.
